src/Rootfs.jl

```diff
@@ -599,8 +606,21 @@ function choose_shards(p::AbstractPlatform;
             preferred_rust_version::VersionNumber = maximum(getversion.(Rust_builds)),
             # Always default to the latest Go version
             preferred_go_version::VersionNumber = maximum(getversion.(Go_builds)),
+            # Always default to the latest OCaml version
+            preferred_ocaml_version::VersionNumber = maximum(getversion.(OCaml_builds)),
         )
 
+    # The preferred GCC version depends on the compilers we are using.
+    if preferred_gcc_version === nothing
+        preferred_gcc_version = if :ocaml in compilers
```

Comment:
So, I think this is ok, but I wanted to point out that we already do some shenanigans in `gcc_version` at https://github.com/JuliaPackaging/BinaryBuilderBase.jl/blob/dd705399548bb909f4c6e5f815b6de917f746faa/src/Rootfs.jl#L447-L538 for example in certain cases we change the version of the gcc toolchain based on the...llvm toolchain.  I was never proud of that mess though.